client/landing/stepper/declarative-flow/internals/steps-repository/sensei-purpose/index.tsx

```diff
@@ -0,0 +1,158 @@
+import { TextControl } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import React, { useEffect } from 'react';
+import DocumentHead from 'calypso/components/data/document-head';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { SenseiStepContainer } from '../components/sensei-step-container';
+import { SenseiStepProgress } from '../components/sensei-step-progress';
+import PurposeItem from './purpose-item';
+import type { Step } from '../../types';
+import './style.scss';
+
+const wait = ( ms: number ) => new Promise( ( res ) => setTimeout( res, ms ) );
+
+export const purposes = [
+	{
+		id: 'sell_courses',
+		label: __( 'Sell courses and generate income' ),
+		plugin: { slug: 'woocommerce', id: 'woocommerce/woocommerce' },
+	},
+	{
+		id: 'provide_certification',
+		label: __( 'Provide certification' ),
+		plugin: {
+			slug: 'sensei-certificates',
+			id: 'sensei-certificates/woothemes-sensei-certificates',
+		},
+	},
+	{
+		id: 'educate_students',
+		label: __( 'Educate students' ),
+		description: null,
+	},
+	{
+		id: 'train_employees',
+		label: __( 'Train employees' ),
+		description: null,
+	},
+];
+
+type FormState = {
+	selected: string[];
+	other: string;
```

Comment:
This is the schema for the Sensei setup wizard step: https://github.com/Automattic/sensei/blob/ab34051a8e7a79a5a5660d4df07364d8ee87568d/includes/admin/class-sensei-setup-wizard.php#L41-L44

I agree about the generic names in principle, but since the component is scoped to just selecting purposes I'd keep it like this to avoid having to map it to a different schema.

Reply:
I don't think that a schema that is defined in a different codebase should affect the implementation details here. I imagine that these two codebases will evolve separately from now on.

In comparison though, the values in the setup wizard are enclosed in a 'purpose' object which makes it easier to understand what the values refer to. Here we name the type FormState which is more generic.

Reply:
Updated the naming a bit in [3aa8fc4](https://github.com/Automattic/wp-calypso/pull/72582/commits/3aa8fc4b86ffa9904eef3fe4de4cb3e49fc5af4e), and also made the selected - wasSelected code a bit clearer.